cli/main.ml

```diff
@@ -187,7 +187,21 @@ let serve_cmd =
   let ignore_checksums =
     let doc = "Do not verify checksums" in
     Arg.(value & flag & info ["ignore-checksums"] ~doc) in
-  Term.(ret(pure Impl.serve $ common_options_t $ source $ source_fd $ source_format $ source_protocol $ destination $ destination_fd $ destination_format $ destination_size $ prezeroed $ progress $ machine $ tar_filename_prefix $ ignore_checksums)),
+  Term.(ret(pure Impl.serve
+            $ common_options_t
+            $ source
+            $ source_fd
+            $ source_format
+            $ source_protocol
+            $ destination
+            $ destination_fd
+            $ destination_format
+            $ destination_size
```

Comment:
The destination size is actually the size of the input file and where we obtain the size if we are parsing a stream. I would like to understand why it is called destination size - if anybody knows. I have not renamed it for now but I think we should.

Reply:
Can you point to where the destination size becomes associated with the input file's size? I'm probably missing something, but this code looked right at a first glance: https://github.com/xapi-project/vhd-tool/blob/master/src/impl.ml#L990-L999

Reply:
This code in xapi:

* https://github.com/xapi-project/xen-api/blob/eab3129c29c2a624c0a3db65410b936a974dc952/ocaml/xapi/vhd_tool_wrapper.ml#L76

Reply:
It is passed in from https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/import_raw_vdi.ml#L83, I think the assumption could be that if we are importing a raw VDI, then after we've imported it we have to end up with a VDI the same size as the one we were importing.
But we only know that if the client sends us a Content-Length field, if it uses chunked encoding for upload then we won't know the final size until we receive the final chunk telling us that we reached the end.